### PR TITLE
Warning fixes & clang-tidy

### PR DESCRIPTION
**Pull request by In-line**
#### Describe your PR, what does it fix/add?

Warning fixes, clang-tidy

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

Be careful while reading patch, some of the fixed parts of the code look suspicious.

#### Is it ready for merging, or does it need work?

Yes, if it passes review

### CODE REVIEW

**Comment by vaxerski** [comment]
I don't like these changes. Why?

**Comment by In-line** [comment]
@vaxerski Because Clang/GCC outputs warnings

**Comment by In-line** [comment]
Well, this are both GCC and Clang and Clang-tidy warnings, I tested with all of them, until there are no issues left.

If you want some parts to be removed, please feel free to tell me. 

As this patch is fairly small in line count, I don't see how it will damage the project. On the other hand it removes spam from build logs and dead code.

**Comment by In-line** [comment]
As patches are separated in self-contained commits, feel free to decide on what subset of the changes you are comfortable with. I'm not sure there is enough technical reasoning to drop all of them.

**Comment by In-line** [comment]
@vaxerski If you don't have any intentions to merge this patches, please let me now, so I can stop keeping them up to date

**Comment by vaxerski** [comment]
I'll give you a review of which patches I am ok with and you can remove those not included, then we can merge

**Review by vaxerski — COMMENTED** [review]
those with ok are to be left, the rest removed, and we can merge

File: src/events/Windows.cpp [hunk 1]
```diff
@@ -150,8 +150,8 @@ void Events::listener_mapWindow(void* owner, void* data) {
         (PWINDOW->m_bIsX11 && PWINDOW->m_uSurface.xwayland->fullscreen);
     bool requestsFakeFullscreen = false;
     bool requestsMaximize       = false;
-    bool overridingNoFullscreen = false;
-    bool overridingNoMaximize   = false;
+    [[maybe_unused]] bool overridingNoFullscreen = false;
+    [[maybe_unused]] bool overridingNoMaximize   = false;
```
[the remark this answers is not in the document]
Comment (aslanyan-a) [reply src/events/Windows.cpp:154]: @vaxerski Btw why? This is standard C++17

**Review by vignesh1507 — COMMENTED** [review]
can you explain me, what have you changed/updated in the code?

File: src/helpers/BezierCurve.cpp [hunk 2]
```diff
@@ -30,8 +30,8 @@ void CBezierCurve::setup(std::vector<Vector2D>* pVec) {
     const auto POINTSSIZE = m_aPointsBaked.size() * sizeof(m_aPointsBaked[0]) / 1000.f;
 
     const auto BEGINCALC = std::chrono::high_resolution_clock::now();
-    for (float i = 0.1f; i < 1.f; i += 0.1f)
-        getYForPoint(i);
+    for (int i = 0; i < 10; i++)
```
[the remark this answers is not in the document]
Comment (Honkazel) [reply src/helpers/BezierCurve.cpp:33]: Already merged
Thanks to Tom

**Comment by Honkazel** [comment]
@In-line So? Would you update this PR or not?

**Comment by In-line** [comment]
Hi, I don't use Hyprland anymore, I switched to KDE.

I'm frustrated with the main dev and I don't intend to maintain or create PRs to Hyprland in the near future.

Feel free to take over, if you're interested.

-------- Original Message --------

**Comment by vaxerski** [comment]
closing then

**Pull request closed without merging**